bin/stop_be.sh

```diff
@@ -83,18 +93,28 @@ if [ -f $pidfile ]; then
     pid=`cat $pidfile`
     pidcomm=`ps -p $pid -o comm=`
     if [ "starrocks_be"x != "$pidcomm"x ]; then
-        echo "ERROR: pid process may not be be. "
+        echo "ERROR: pid process may not be BE"
+        exit 1
+    fi
+    
+    kill -${SIG} $pid > /dev/null 2>&1
+    if [ $? -ne 0 ]; then
+        rm $pidfile
         exit 1
     fi
 
-    if kill -0 $pid >/dev/null 2>&1; then
-        kill -${sig} $pid > /dev/null 2>&1
-        if [ $? -ne 0 ]; then
-            exit 1
-        fi
-        while kill -0 $pid >/dev/null 2>&1; do
+    # Waiting for a process to exit
+    start_ts=$(date +%s)
+    while kill -0 $pid > /dev/null 2>&1; do
+        if [ $(($(date +%s) - $start_ts)) -gt $TIME_OUT ]; then
```

Comment:
provide the ability to disable this timeout in test env. so if there is any unexpected hang, it is still possible to keep the env for troubleshooting.

Reply:
Instead of adding a test environment variable, Is it better to specify a very large timeout? In any case, you have to make special configuration for the test environment.

Reply:
it will break the configuration `loop_count_wait_fragments_finish = x` if the timeout is shorter than 10 * loop_count_wait_fragments_finish, the script won't wait long enough before BE process decides to exit.

Reply:
I think it is not a conflict, but a priority issue. It is possible that BE cannot be stopped due to other reasons. stop_be.sh controls all timeouts and is a fallback measure.

Reply:
I am not questioning the timeout overrule the configuration in BE, but question the default behavior if the configuration is set to be.conf but no explicit timeout provided in stop_be.sh with -g. will it be proper to terminate the BE process with the given default timeout?

Reply:
your current implementation is

`stop_be.sh -g`, no matter what value configured in be.conf, the process will be terminated in 60s (the default timeout), if a longer graceful period is needed,  timeout must be provided as `stop_be.sh -g --timeout xxx`.

My proposal is
`stop_be.sh -g` should be no timeout, let the backend process to determine when to exit.
`stop_be.sh -g --timeout xxx`, the script will take action if backend process can't be exit in xxx seconds, (overrule the behavior of be process configuration).